sygnal/gcmpushkin.py

```diff
@@ -465,8 +465,12 @@ def __init__(self):
             reg_id (str): a registration ID
             canonical_reg_id (str): the canonical registration ID for `reg_id`
         """
-        await self.db.runQuery(
-            "INSERT OR REPLACE INTO gcm_canonical_reg_id VALUES (?, ?);",
+        await self.db.runOperation(
+            """
+            INSERT INTO gcm_canonical_reg_id VALUES (%s, %s)
+            ON CONFLICT (reg_id) DO UPDATE
+                 SET canonical_reg_id = EXCLUDED.canonical_reg_id;
+            """,
```

Comment:
Does `runOperation` `commit()` automatically or do we need to do so explicitly here as well?

Reply:
`runOperation` commits automatically